docs/core/integration_diagnostics.md

```diff
@@ -0,0 +1,50 @@
+---
+title: Integration diagnostics
+sidebar_label: "Diagnostics"
+---
+
+Integrations can provide diagnostics to help the user gather data to aid in troubleshooting. Diagnostics can be provided for config entries but also individually for each device entry.
+
+Users can download config entry diagnostics from the config entry options menu, on the integration page. For device diagnostics, users can download them from the device menu.
+
+:::warning
+It is critical to ensure that no sensitive data is exposed. This includes but is not limited to:
+- Passwords and API keys
+- Authentication tokens
+- Location data
+- Personal information
+- Device identifiers
```

Comment:
```suggestion
```
In general we don't see a device identifier as sensitive. In general they are not sensitive, and I think this is a rule that should be applied differently per integration, since there probably are integrations where its sensitive, in those cases we should indeed redact them. But for now, they are amazing to identify the right data in the diagnostics